lte/gateway/python/magma/pipelined/app/enforcement_stats.py

```diff
@@ -472,13 +468,13 @@ def _handle_flow_stats(self, stats_msgs):
             return
         # Send report even if usage is empty. Sessiond uses empty reports to
         # recognize when flows have ended
-        self._report_usage(current_usage)
+        self._report_usage(current_usage[1])
```

Comment:
?

Reply:
For current implementation, local_f_teid_ng value is only distingues the 4G or 5G flows.
The usage report is sending to sessiond using grpc methods are different for 4G and 5G.
The class object are diffenet for 4G and 5G for session creation/storing/termination.

If we are sending all flows to both then sessiond of 4g object will start for cleanup all dangling flows. That all belongs to 5g flows.
and vice-versa.

current_usage = defaultdict(lambda: defaultdict(RuleRecord))
Here, We defined list of defaultdict for index 0 belongs to 5g flows and for index 1 belongs to 4g flows.

Reply:
This seems hacky, we can either 
 - tag the RuleRecord with 5g flag if that helps?
 - do 2 separate report calls but lets rename it to current_usage/current_usage_5g or something.  
 
 @uri200 @themarwhal  what would you prefer on sessiond side?

Reply:
@koolzz
 1) 
message RuleRecord {
  // IMSI. This value along with teid uniquely identifies a session.
  string sid = 1;
  string rule_id = 2;
  uint64 bytes_tx = 3;
  uint64 bytes_rx = 4;
  // Used for HE
  string ue_ipv4 = 5;
  // Used for HE
  string ue_ipv6 = 6;
  uint64 dropped_tx = 7;
  uint64 dropped_rx = 8;
  // Rule version with which the flow was installed with
  uint64 rule_version = 9;
  // Uplink TEID
  uint32 teid = 10;
  //Indicates that the report is final and the flow will be deleted
  bool final_record = 11;
}

teid parameter is only used by 5G as of now, We can use this variable. But 
We need to put the check for teid based RuleRecord will be ignored in LocalEnforcer for 4G, 
Otherwise 4g object will start for cleanup all dangling flows.

If we will put the below check in LocalEnforcer.cpp then not required above changes in pipeliend.
 
const uint32 teid = record.teid();
if teid:
   continue;
   
2)  We start off a thread to periodically poll stats from Pipelined in sessiond level if "enable_pull_stats" is true.
    periodic_stats_requester->start_loop(local_enforcer, interval);
   So, Not a good approaches for start new loop for 5G.

@uri200 , @themarwhal , please suggest your view.

Reply:
Also let's not differentiate 5g vs not just based on teid as we might soon add that to the 4g flows. Let's explicitly add the 5g flag if needed

Reply:
I do not have enough context on pipelined side. But the question is: are 4g or 5g flows different? 

Why is the usage report sent using different GRPC call?  To my understanding reporting traffic is the same, no matter if this is 4g, 5g or wifi